14 there should be multiple end products produced by factories or buildings

NoelWO opened the pull request.

NoelWO committed:
Finished

NoelWO committed:
Bug fix

NoelWO committed:
Improved texture

linuskar commented:
Could you update from main and resolve the conflicts

NoelWO committed:
Merge branch 'main' into 14-there-should-be-multiple-end-products-produced-by-factories-or-buildings

linuskar commented:
Remake your selectable scenes for the new buildings, there are missing nodes from the template

NoelWO committed:
Deleted

NoelWO committed:
Delete again

NoelWO committed:
Hopefully fixes

NoelWO committed:
Merge branch 'main' into 14-there-should-be-multiple-end-products-produced-by-factories-or-buildings

NoelWO committed:
More asset fixes

Heldereim commented on resources/buildings/coal_mine.tres:
```diff
@@ -21,7 +21,7 @@ input_types = Array[int]([])
 output_types = Array[int]([1, 3])
 building_type = 2
 valid_tile_types_to_place_on = Array[int]([1, 2, 3, 4])
-building_size = Vector2(1, 1)
```
If you are changing the size of a building you should also update the size of the clickable in the scene. This has not been done. Instead the edges of the coal mine do not respond to clicks. Please update the size of the clickable.

Heldereim commented on resources/buildings/iron_mine.tres:
```diff
@@ -21,7 +21,7 @@ input_types = Array[int]([])
 output_types = Array[int]([0, 3])
 building_type = 1
 valid_tile_types_to_place_on = Array[int]([1, 2, 3, 4])
-building_size = Vector2(1, 1)
```
Similar to the coal mine, please update the clickables. This should be done in the coal mine scene and iron mine scene respectively.

Heldereim commented on scenes/buildings/gathering_buildings/iron_mine.tscn:
As written above, you need to change the 'Clickable' node in this scene to be as large as the image it is part of.

Heldereim commented on scenes/buildings/gathering_buildings/coal_mine.tscn:
As written above, you need to change the 'Clickable' node in this scene to be as large as the image it is part of.

Heldereim commented:
There seems to be an issue with the building select list?
It looks like the selectables are shifted one step (Road name for saw mill, Saw Mill name for Steel Mill and so on)
![bild_2025-04-19_090700342](https://github.com/user-attachments/assets/2a4de023-b7a2-4333-9445-8f182ebe993d)

Heldereim commented:
Additionally, the blueprints have seemingly not been added to the build manager script leading to a crash as soon as one tries to build one of the three new buildings (Saw Mill, Steel Mill, Gear Factory). To do this go to the game scene, click on build manager and add the blueprints for the new buildings on the right hand side.
![image](https://github.com/user-attachments/assets/6fdfb1e2-2838-4caf-a311-3c277d54da57)
![image](https://github.com/user-attachments/assets/07e76947-7e1a-45f2-8bad-fc58796a578b)

NoelWO committed:
Bug fixes

NoelWO committed:
Merge branch 'main' into 14-there-should-be-multiple-end-products-produced-by-factories-or-buildings

NoelWO committed:
Smoke bug

Heldereim committed:
Merge branch 'main' into 14-there-should-be-multiple-end-products-produced-by-factories-or-buildings

Heldereim committed:
Merge branch 'main' into 14-there-should-be-multiple-end-products-produced-by-factories-or-buildings

NoelWO committed:
Merge branch 'main' into 14-there-should-be-multiple-end-products-produced-by-factories-or-buildings

NoelWO committed:
Merge branch 'main' into 14-there-should-be-multiple-end-products-produced-by-factories-or-buildings

linuskar reviewed (changes requested):
- Make sure the smoke particles are not emitted always for buildings, for example like the coal mine or coal power plant. Fix this for all buildings that have smok emitters. Make sure they have emitting to not on, not checked.
- ![image](https://github.com/user-attachments/assets/2771b78a-e0a0-4a66-a0a5-c0928684f876)
- Also some buildings are missing smoke emitters for the amount of smoke towers they have in the sprite
- Some buildings miss clickable buttons for building info
- Update and make sure the building data is reasonable for the new buildings. Some of them miss max storage and what resources they output for example.

NoelWO committed:
more fixes

Heldereim committed:
Extendable resource ui, bugfix

Made the resource ui extendable. Created a generic resource ui entry. Overrode the output overflow check for gathering buildings.

Heldereim committed:
Fixed saw mill and steel mill

Heldereim committed:
Fixed gear factory

Heldereim committed:
Fixed size of clickabe for coal mine and iron mine

Heldereim committed:
Merge branch 'main' into 14-there-should-be-multiple-end-products-produced-by-factories-or-buildings

Heldereim committed:
Build list closed by default

Changed the building select list to be closed by default.

Heldereim committed:
Smoke generation

Updated the smoke emitter connections for saw mill, gear factory and steel mill. Updated the functions for checking if smoke should be generated.

linuskar reviewed (changes requested):
Could the output of the steel mill be configured. Right now it wont reach the max capacity of 100 steel because it outputs 6 steel.

Heldereim committed:
Steel mill max storage

Updated the max storage for steel in the steel mill to 120 steel to make it so it becomes completely filled when producing.
